### ISSUE TITLE
JSON Expressions Don't work within Case  Expressions

### ISSUE THREAD

**Issue by zeeshanshabbir93** [lf-edge/ekuiper#824]
I am trying to join a schema-less table "digitalll" with a stream "tagStream".  Rule is as follows 

`SELECT digitalll.payload->thresholdLOW as debugMsg, CASE 
    WHEN tagStream.v > 60 THEN "HIHI" 
    WHEN tagStream.v > cast(digitalll.payload->thresholdLOW,"float") THEN "LOWLOW"
    END as condition
FROM tagStream INNER JOIN digitalll on tagStream.eid = digitalll.eid`

Data for tagStream is {"eid":"eid1","id":"tag1","t":1616079610241,"v":39} 
Data for digitalll table is {"eid":"eid1","id":"tag1","payload":{"thresholdHI":55,"thresholdLOW":40}}

When tagStream.v is greater than 40 then only "debugMsg" attribute is received in the output, when it is less than 40 no message is received.

Also, JSON expressions don't work on either side of expression as well as after "THEN"

I am using  **emqx/kuiper:1.2.0-alpine** image.

**Comment by ngjaying** [comment]
@zeeshanshabbir93 Thanks for opening the issue. I can reproduce the problem. It is actually a problem of alias.

Currently, you can use a workaround by removing the alias.
```
SELECT digitalll.payload->thresholdLOW as debugMsg, CASE WHEN tagStream.v > 60 THEN "HIHI" WHEN tagStream.v > cast(digitalll.payload->thresholdLOW,"float") THEN "LOWLOW" END FROM tagStream INNER JOIN digitalll on tagStream.eid = digitalll.eid
```
Your condition will get a default field name `kuiper_field_0`. You can use datatemplate to rename the output field name.

**Comment by zeeshanshabbir93** [comment]
Thanks.

**Comment by ngjaying** [comment]
Verified. Will be released at 1.3.0
